### DISCUSSION TITLE
How can I decrease image size as low as possible with resize?

### DISCUSSION THREAD

**Discussion by martonx** [SixLabors/ImageSharp#1521] [Q&A] [answered]
Dear Supporters,

I have this image: 
![97-Veinoplus-visszerkezelo-keszulek-elektroda-pozicio](https://user-images.githubusercontent.com/407433/105825615-170cb380-5fc0-11eb-9fee-3c2ce90ce9f2.jpg)

And I use this code to resize it:

```C#
public byte[] ResizeImage(byte[] imageByteArray, int? maxHeight = null, int? maxWidth = null, string imageFormat = "image/jpeg")
  {
      using (var original = Image.Load(imageByteArray))
      {
          int newWidth = original.Width;
          int newHeight = original.Height;

          if (maxWidth < original.Width || maxHeight < original.Height)
          {
              var ratioX = (double)(maxWidth ?? original.Width) / original.Width;
              var ratioY = (double)(maxHeight ?? original.Height) / original.Height;
              var ratio = Math.Min(ratioX, ratioY);

              newWidth = (int)(original.Width * ratio);
              newHeight = (int)(original.Height * ratio);
          }

          original.Mutate(x => x.Resize(newWidth, newHeight));

          using (var ms = new MemoryStream())
          {
              switch (imageFormat)
              {
                  case "image/jpeg":
                      original.SaveAsJpeg(ms);
                      break;
                  case "image/png":
                      original.SaveAsPng(ms);
                      break;
                  case "image/gif":
                      original.SaveAsGif(ms);
                      break;
                  default:
                      throw new NotSupportedException($"Image format {imageFormat} not supported!");
              }

              return ms.ToArray();
          }
      }
  }
```

ImageByteArray is the linked image in bytes. Maxheight is 55, and MaxWidth is 55 also. Finally from the original 640*480 ~700kbyte image, I got an 55*41 ~560kbyte image. How the hell can I resize this image to a few kbyte size?

**Comment by antonfirsov** [comment]
@martonx if `55 x 41 px` indeed results in 560 kbyte, that would seem very unexpected to me. Are you sure you are not overlooking something and the resulting output is not bigger actually?

Can you try a simplified code, which takes an input jpeg and resizes it exactly to 50x50, and check the output size?

**Comment by JimBobSquarePants** [comment, marked as answer]
It's got a massive ICC profile embedded in it.

![image](https://user-images.githubusercontent.com/385879/105831620-db251e80-5fbe-11eb-9861-e00200662b0a.png)


Using the following code to resize and strip the profile.

```c#
using (var image = Image.Load(Path.Combine(inPath, "discussions.jpg")))
{
    var options = new ResizeOptions { Mode = ResizeMode.Max, Size = new Size(55, 55) };
    image.Mutate(x => x.Resize(options));
    image.Metadata.IccProfile = null;
    image.Save(Path.Combine(outPath, "discussions-no-meta.jpg"));
}
```

A jpeg of 55x41px is 3.07kb

![discussions-nometa](https://user-images.githubusercontent.com/385879/105831265-68b43e80-5fbe-11eb-81e5-442ecd09da86.jpg)

**Reply by martonx** [reply]
WOW, you are awsome, this image.Metadata.IccProfile = null; saves my day! Thank you very much!

**Reply by JimBobSquarePants** [reply]
Happy to help!
